tests/integration/new_relations/test_relations_coherence.py

```diff
@@ -0,0 +1,127 @@
+                # was created for the application.
+                cursor.execute("DROP TABLE IF EXISTS test;")
+                cursor.execute("CREATE TABLE test(data TEXT);")
+                cursor.execute("INSERT INTO test(data) VALUES('some data');")
+                cursor.execute("SELECT data FROM test;")
+                data = cursor.fetchone()
+                assert data[0] == "some data"
+        connection.close()
+
+        await ops_test.model.applications[DATABASE_APP_NAME].remove_relation(
+            f"{DATABASE_APP_NAME}:database",
+            f"{APPLICATION_APP_NAME}:{FIRST_DATABASE_RELATION_NAME}",
+        )
+        await ops_test.model.wait_for_idle(apps=[DATABASE_APP_NAME], status="active", timeout=1000)
+
+        # Relate with client-app 2nd time, wait-for-ready, check relation data
+        await ops_test.model.add_relation(
+            f"{APPLICATION_APP_NAME}:{FIRST_DATABASE_RELATION_NAME}", DATABASE_APP_NAME
+        )
+        await ops_test.model.wait_for_idle(apps=APP_NAMES, status="active")
+
+        primary_connection_string = await build_connection_string(
+            ops_test, APPLICATION_APP_NAME, FIRST_DATABASE_RELATION_NAME
+        )
+        with psycopg2.connect(primary_connection_string) as connection:
+            connection.autocommit = True
+            with connection.cursor() as cursor:
+                cursor.execute("SELECT data FROM test;")
+                data = cursor.fetchone()
+                assert data[0] == "some data"
+        connection.close()
+
+        await ops_test.model.applications[DATABASE_APP_NAME].remove_relation(
+            f"{DATABASE_APP_NAME}:database",
+            f"{APPLICATION_APP_NAME}:{FIRST_DATABASE_RELATION_NAME}",
+        )
+        await ops_test.model.wait_for_idle(apps=[DATABASE_APP_NAME], status="active", timeout=1000)
+
+        # Repeat re-relation using all available relation options (e.g. legacy_roles)
+        await ops_test.model.applications[APPLICATION_APP_NAME].set_config({
+            "legacy_roles": "true",
+        })
+        await ops_test.model.wait_for_idle(
+            apps=[APPLICATION_APP_NAME], status="active", timeout=1000
+        )
+
+        await ops_test.model.add_relation(
+            f"{APPLICATION_APP_NAME}:{FIRST_DATABASE_RELATION_NAME}", DATABASE_APP_NAME
+        )
+        await ops_test.model.wait_for_idle(apps=APP_NAMES, status="active")
+
+        primary_connection_string = await build_connection_string(
+            ops_test, APPLICATION_APP_NAME, FIRST_DATABASE_RELATION_NAME
+        )
+        connection = psycopg2.connect(primary_connection_string)
+        connection.autocommit = True
+        cursor = connection.cursor()
+        random_name = f"test_{''.join(secrets.choice(string.ascii_lowercase) for _ in range(10))}"
+        cursor.execute(f"CREATE DATABASE {random_name};")
+        cursor.execute(f"DROP DATABASE {random_name};")
+        connection.close()
```

Comment:
I don't think we can create DBs with the test app user. This should most probably be creating tables instead.

Reply:
+1 to @dragomirp , bit I am surprised to see [it passed](https://github.com/canonical/postgresql-operator/actions/runs/8812966437/job/24189993180#step:21:167).

Reply:
The test app has permissions for that: https://github.com/canonical/postgresql-test-app/blob/3edafc10b4b5aab29355820924b39c208bec76b1/src/charm.py#L35C1-L35C17